### PR TITLE
Fix/bs 8234 tenant session timeout

### PR DESCRIPTION
**Pull request by cpuy**
Opened on `tasklist` branch since I would like to test it in real bundle while it is merged.
Please tell me while it has been merged.

I had to do some refactor so PR is split in 3 commits with third one that fix the bug.

On Engine log in, we add tenant id in cookie, while creating login url we had tenant id in url from parameters if it is set, from cookie otherwise

Please merge associated SP PR https://github.com/bonitasoft/bonita-web-sp/pull/62

### CODE REVIEW

File: server/src/main/java/org/bonitasoft/console/common/server/auth/impl/standard/StandardAuthenticationManagerImpl.java [hunk 1]
```diff
@@ -48,13 +45,22 @@ public String getLoginPageURL(final HttpServletRequestAccessor request, final St
...
     }
 
+    private String getTenantId(HttpServletRequestAccessor request) {
+        String tenantId = request.getTenantId();
+        if (tenantId == null) {
+            tenantId = PortalCookies.getTenantCookieFromRequest(request.asHttpServletRequest());
+        }
+        return tenantId;
+    }
+
```
Comment (abirembaut) [inline server/src/main/java/org/bonitasoft/console/common/server/auth/impl/standard/StandardAuthenticationManagerImpl.java]: I think the same should be done in other AuthenticationManager implementations  in bonita-web-sp/pull/62

Comment (cpuy) [inline server/src/main/java/org/bonitasoft/console/common/server/auth/impl/standard/StandardAuthenticationManagerImpl.java]: Yep I've changed only the StandardAuthenticationManager

AFAIK there is only CASRemoteAuthenticationManagerImpl in bonita-web-sp, I'll check with @redboul if we need to change it also

Comment (abirembaut) [inline server/src/main/java/org/bonitasoft/console/common/server/auth/impl/standard/StandardAuthenticationManagerImpl.java]: Yep, you are right. There is also JAASAuthenticationManagerImpl in bonita-web but it doesn't look like it needs to be changed.
IMHO, CASRemoteAuthenticationManagerImpl needs to be changed

**Comment by cpuy** [comment]
I've changed implementation as follow
* retrieve tenant from cookie in `TenantIdAccessor` and use this accessor in each AuthentManager.
* JaasAuthentManager now extends standardAuthentmanager to remove code duplication
* Also updated CASRemoteAuthenticationManagerImpl in https://github.com/bonitasoft/bonita-web-sp/pull/62

**Pull request merged**